api/tests/routes/adage_iframe/get_categories_test.py

```diff
@@ -0,0 +1,97 @@
+from typing import ByteString
+from typing import Optional
+
+import pytest
+
+from tests.routes.adage_iframe.utils_create_test_token import create_adage_jwt_fake_valid_token
+
+
+def _create_adage_valid_token_with_email(
+    email: str,
+    civility: Optional[str] = "Mme",
+    lastname: Optional[str] = "LAPROF",
+    firstname: Optional[str] = "Jeanne",
+    uai: Optional[str] = "EAU123",
+) -> ByteString:
+    return create_adage_jwt_fake_valid_token(
+        civility=civility, lastname=lastname, firstname=firstname, email=email, uai=uai
+    )
+
+
+pytestmark = pytest.mark.usefixtures("db_session")
+
+
+class Returns200Test:
+    def test_get_categories(self, client):
+        # Given
+        adage_jwt_fake_valid_token = _create_adage_valid_token_with_email(email="toto@mail.com", uai="12890AI")
+        client.auth_header = {"Authorization": f"Bearer {adage_jwt_fake_valid_token}"}
+
+        # When
+        response = client.get("/adage-iframe/offers/categories")
+
+        assert response.status_code == 200
+        assert list(response.json.keys()) == ["categories", "subcategories"]
+        assert all(
+            list(subcategory_dict.keys())
```

Comment:
Bien que depuis python 3.6, il semblerait que les clés d'un dictionnaire soient ordonnées dans le sens d'insertion, je pense qu'on ne veut pas tester l'ordre des clés ici.
Si quelqu'un modifie l'ordre des clés dans le dict ALL_CATEGORIES par exemple, ce test sautera.
Il y a plusieurs moyens d'y arriver, je ne sais pas lequel est le plus propre entre : 
- convertir les deux côtés en `set()` et tester l'égalité
- checker dans une boucle for que chaque clé est dans la liste
- appliquer `sorted()` sur les deux listes à comparer

@cgaunet tu préfères quoi?

Reply:
ce serait à modifier aussi côté routes/pro car j'ai repris le code de là

Reply:
Je suis aligné avec Gautier, chaud si tu peux corriger aussi côté routes/pro parce que c'est très flaky :)